scrape_s3_redirects.py

```diff
@@ -0,0 +1,79 @@
+import argparse
+import boto3
+import datetime
+import time
+import json
+
+
+def extractKey(s3_object)-> list:
+    return s3_object.key
+
+def writeObjectKeys(bucket)-> list:
+    bucketObj = boto3.session.Session().resource("s3").Bucket(bucket)
+    s3_objects = list(bucketObj.objects.all())
+    keys = { bucket: list(map(extractKey, s3_objects))}
+    print(f"Number of objects in bucket: {len(s3_objects)}")
+    f = open(f"scraped-redirects/keys.txt", 'a')
+    f.write('\n')
+    f.write(json.dumps(keys))
+    return keys
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--bucket", help = "Which bucket to to look in", type = str)
+    parser.add_argument("--firstIndex", help = "First index for objects", type = int)
+    parser.add_argument("--lastIndex", help = "Last index for objects", type = int)
+    parser.add_argument("--redirectFileNumber", help = "Skip generation of a list of object keys", type = bool)
+  
+
+    args = parser.parse_args()
+    bucket = args.bucket
+
+    start_time = time.time()
+    now = datetime.datetime.now()
+    print(f"Current date and time using datetime: {now}")
+
+
+    f = open(f"scraped-redirects/keys.txt")
+    keys_file = json.load(f)
+    print(keys_file[bucket][0])
+    if (keys_file[bucket]):
+        keys = keys_file[bucket]
```

Comment:
referring to [this thread](https://github.com/mongodb/snooty-create-redirects/pull/1/files#r1989487901), but this will have outdated keys if using the committed ones. i think this should update it on the fly WDYT?

bonus if it can make the whole key file on the fly as well

Reply:
Sorry, could you elaborate on what you're suggesting?

Reply:
ah sorry meant this comment https://github.com/mongodb/snooty-create-redirects/pull/1#discussion_r1989487901

Wanted to always refresh the keys per bucket, instead of relying on file. (or gitignore the file so its always fresh)

Reply:
I .gitignored the bucket so that it's always fresh when you clone, but it's really helpful for running the script to have the same keys stored in the file in the same order because you'll need to run it multiple times because there are so many objects to look thorugh